WordPress/Classes/ViewRelated/Blaze/BlazeWebViewController.swift

```diff
@@ -0,0 +1,124 @@
+import UIKit
+import WebKit
+
+class BlazeWebViewController: UIViewController, BlazeWebView {
+
+    // MARK: Private Variables
+
+    private let webView: WKWebView
+    private var viewModel: BlazeWebViewModel?
+    private let progressView = WebProgressView()
+
+    // MARK: Lazy Loaded Views
+
+    private lazy var cancelButton: UIBarButtonItem = {
+        let button = UIBarButtonItem(title: Strings.cancelButtonTitle,
+                                     style: .plain,
+                                     target: self,
+                                     action: #selector(cancelButtonTapped))
+        return button
+    }()
+
+    // MARK: Initializers
+
+    init(source: BlazeWebViewCoordinator.Source, blog: Blog, postID: NSNumber?) {
+        self.webView = WKWebView(frame: .zero)
+        super.init(nibName: nil, bundle: nil)
+        viewModel = BlazeWebViewModel(source: source, blog: blog, postID: postID, view: self)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: View Lifecycles
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.isModalInPresentation = true
+        overrideUserInterfaceStyle = .light
+        configureSubviews()
+        configureWebView()
+        configureNavBar()
+        viewModel?.startBlazeFlow()
+    }
+
+    // MARK: Private Helpers
+
+    private func configureSubviews() {
+        let subviews = [progressView, webView]
+        let stackView = UIStackView(arrangedSubviews: subviews)
+        stackView.axis = .vertical
+        view = stackView
+    }
+
+    private func configureWebView() {
+        webView.navigationDelegate = self
+        webView.customUserAgent = WPUserAgent.wordPress()
```

Comment:
It makes sense to define a new user agent for Jetpack, but that needs to be a coordinated effort with the dotcom team, as the endpoints need to be able to identify the new user agent. This will probably need its own project.